oxidation/crates/parsec_api_crypto/src/rustcrypto/secret.rs

```diff
@@ -35,23 +39,19 @@ impl SecretKey {
 
     pub fn decrypt(&self, ciphered: &[u8]) -> Result<Vec<u8>, &'static str> {
         let cipher = XSalsa20Poly1305::new(&self.0);
-        let nonce_slice = ciphered.get(..NONCE_SIZE).ok_or("Invalid data size")?;
+        let nonce_slice = ciphered
+            .get(..NONCE_SIZE)
+            .ok_or("The nonce must be exactly 24 bytes long")?;
         cipher
             .decrypt(nonce_slice.into(), &ciphered[NONCE_SIZE..])
             .map_err(|_| "Decryption error")
     }
 
-    // TODO
-    pub fn hmac(&self, _data: &[u8], _digest_size: usize) -> Vec<u8> {
-        // // blake2b(data, digest_size=digest_size, key=self, encoder=RawEncoder)
-        // let key = blake2b::Key::from_slice(&self.0);
-        // blake2b::derive_from_key(
-        //     subkey: &mut [u8],
-        //     subkey_id: u64,
-        //     ctx: [u8; 8],
-        //     key,
-        // );
-        vec![]
+    pub fn hmac(&self, data: &[u8]) -> Vec<u8> {
```

Comment:
why remove the `digest_size` param ? it is used in the Python implementation

in theory we could truncate the returned vec to the size we want, but
1) it feels very hacky to do that in the caller and it may reduce hash randomness in a subtle way (for instance if the  underlying algorithm is later changed)
2) blake2 handling of `digest_size` is not as simple as trunking the output, so better not play smarter than him...